src/Microsoft.Azure.WebJobs.Host/Tables/TableAttributeBindingProvider.cs

```diff
@@ -68,106 +62,81 @@ private TableAttributeBindingProvider(INameResolver nameResolver, IStorageAccoun
 
             return new ResolvedTableAttribute(attrResolved, client);
         }
-        
+
         public static IBindingProvider Build(INameResolver nameResolver, IConverterManager converterManager, IStorageAccountProvider accountProvider, IExtensionRegistry extensions)
         {
             var original = new TableAttributeBindingProvider(nameResolver, accountProvider, extensions);
 
-            converterManager.AddConverter<JObject, ITableEntity, TableAttribute>(original.JObjectToTableEntityConverterFunc);
+            converterManager.AddConverter<JObject, ITableEntity, TableAttribute>(original.JObjectToTableEntityConverterFunc);            
+            converterManager.AddConverterBuilder<object, ITableEntity, TableAttribute>(typeof(ObjectToITableEntityConverter<>));
 
+            // IStorageTable --> IQueryable<ITableEntity>
+            converterManager.AddConverterBuilder<IStorageTable, IQueryable<OpenType>, TableAttribute>(typeof(TableToIQueryableConverter<>));
+            
             var bindingFactory = new BindingFactory(nameResolver, converterManager);
-            var bindAsyncCollector = bindingFactory.BindToAsyncCollector<TableAttribute, ITableEntity>(original.BuildFromTableAttribute, null, original.CollectAttributeInfo);
 
-            var bindToJobject = bindingFactory.BindToExactAsyncType<TableAttribute, JObject>(original.BuildJObject, null, original.CollectAttributeInfo);
-            var bindToJArray = bindingFactory.BindToExactAsyncType<TableAttribute, JArray>(original.BuildJArray, null, original.CollectAttributeInfo);
+            // Includes converter manager, which provides access to IQueryable<ITableEntity>
+            var bindToExactCloudTable = bindingFactory.BindToInput<TableAttribute, CloudTable>(true, typeof(JObjectBuilder))
+                .SetPostResolveHook<TableAttribute>(original.ToParameterDescriptorForCollector, original.CollectAttributeInfo);
+
+            var bindToExactTestCloudTable = bindingFactory.BindToInput<TableAttribute, IStorageTable>(true, typeof(JObjectBuilder))
+                .SetPostResolveHook<TableAttribute>(original.ToParameterDescriptorForCollector, original.CollectAttributeInfo);
```

Comment:
Can we remove the SetPostResolveHook method and supporting infrastructure, and instead have the builder Type passed in implement an interface providing the hook impls?